client/src/templates/Challenges/quiz/show.tsx

```diff
@@ -57,52 +57,106 @@ interface ShowQuizProps {
   openCompletionModal: () => void;
-  openHelpModal: () => void;
   pageContext: {
     challengeMeta: ChallengeMeta;
   };
-  t: TFunction;
   updateChallengeMeta: (arg0: ChallengeMeta) => void;
   updateSolutionFormValues: () => void;
 }
 
-interface ShowQuizState {
-  hasSubmitted: boolean;
-  quiz: null;
-}
+// Shuffle array using the Fisher–Yates shuffle algorithm
+const shuffle = (arrToShuffle: Array<{ label: ReactNode; value: number }>) => {
+  const arr = [...arrToShuffle];
+
+  for (let i = arr.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [arr[i], arr[j]] = [arr[j], arr[i]];
+  }
+
+  return arr;
+};
+
+const ShowQuiz = ({
+  challengeMounted,
+  data: {
+    challengeNode: {
+      challenge: {
+        fields: { tests },
+        title,
+        description,
+        challengeType,
+        helpCategory,
+        superBlock,
+        block,
+        translationPending,
+        quizzes
+      }
+    }
+  },
+  pageContext: { challengeMeta },
+  initTests,
+  updateChallengeMeta,
+  openCompletionModal,
+  isChallengeCompleted
+}: ShowQuizProps) => {
+  const { t } = useTranslation();
+  const { nextChallengePath, prevChallengePath } = challengeMeta;
+  const container = useRef<HTMLElement | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  // Campers are not allowed to change their answers once the quiz is submitted.
+  // `hasSubmitted` is used as a flag to disable the quiz.
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+
+  // `isPassed` is used as a flag to conditionally render the test or submit button.
+  const [isPassed, setIsPassed] = useState(false);
+
+  const blockNameTitle = `${t(
+    `intro:${superBlock}.blocks.${block}.title`
+  )} - ${title}`;
 
-// Component
-class ShowQuiz extends Component<ShowQuizProps, ShowQuizState> {
-  static displayName: string;
-  private container: React.RefObject<HTMLElement> = React.createRef();
+  // `random` needs to be a ref object so that it won't change between renders.
+  const random = useRef(Math.floor(Math.random() * quizzes.length));
+  const quiz = quizzes[random.current].questions;
+
+  // Initialize the data passed to `useQuiz`
+  const initialQuizData = quiz.map(question => {
+    const distractors = question.distractors.map((distractor, index) => {
+      return {
+        label: (
+          <PrismFormatted className='quiz-answer-label' text={distractor} />
+        ),
+        value: index + 1
+      };
+    });
 
-  constructor(props: ShowQuizProps) {
-    super(props);
-    this.state = {
-      hasSubmitted: false,
-      quiz: null
+    const answer = {
+      label: (
+        <PrismFormatted className='quiz-answer-label' text={question.answer} />
+      ),
+      value: 4
     };
 
-    this.handleSubmit = this.handleSubmit.bind(this);
-  }
+    return {
+      question: <PrismFormatted text={question.text} />,
+      answers: shuffle([...distractors, answer]),
```

Comment:
We can't shuffle here, since it shuffles each render. It makes the UX a bit trippy.

Instead, we can use the same trick as for random: wrap the whole thing in `useState` so it's randomized, but only once.